sdk/ai/ai-projects/test/public/node/finetuning/finetuning.spec.ts

```diff
@@ -0,0 +1,974 @@
+          modelType,
+        );
+        break;
+      case FineTuningJobType.DPO_JOB_TYPE:
+        fineTuningJob = await createDpoFinetuningJob(
+          trainingFile.id,
+          validationFile.id,
+          trainingType,
+          modelType,
+        );
+        break;
+      case FineTuningJobType.RFT_JOB_TYPE:
+        fineTuningJob = await createRftFinetuningJob(
+          trainingFile.id,
+          validationFile.id,
+          trainingType,
+          modelType,
+        );
+        break;
+      default:
+        throw new Error(`Unsupported job type: ${jobType}`);
+    }
+    console.log(
+      `Created fine-tuning ${jobType} ${modelType} ${trainingType} job with ID:`,
+      fineTuningJob.id,
+    );
+
+    console.log(`\nCancelling fine-tuning job with ID: ${fineTuningJob.id}`);
+    const cancelledJob = await openai.fineTuning.jobs.cancel(fineTuningJob.id);
+    console.log(
+      `Successfully cancelled fine-tuning job: ${cancelledJob.id}, Status: ${cancelledJob.status}`,
+    );
+
+    // Validate the cancelled job
+    validateFineTuningJob(cancelledJob, fineTuningJob.id);
+    assert.equal(cancelledJob.status, "cancelled");
+    if (cancelledJob.training_file !== undefined) {
+      assert.equal(cancelledJob.training_file, trainingFile.id);
+    }
+    if (cancelledJob.validation_file !== undefined) {
+      assert.equal(cancelledJob.validation_file, validationFile.id);
+    }
+
+    // Validate method type
+    assert.isNotNull(cancelledJob.method, `Method should not be None for ${jobType} job`);
+    if (cancelledJob.method?.type !== undefined) {
+      assert.equal(cancelledJob.method?.type, expectedMethodType);
+    }
+    console.log(
+      `finetuning cancel ${jobType} ${modelType} ${trainingType} method validation passed - type: ${cancelledJob.method?.type}`,
+    );
+
+    // Verify cancellation persisted by retrieving the job
+    const retrievedJob = await openai.fineTuning.jobs.retrieve(fineTuningJob.id);
+    console.log(
+      `finetuning cancel ${jobType} ${modelType} ${trainingType} verified cancellation persisted for job: ${retrievedJob.id}`,
+    );
+    validateFineTuningJob(retrievedJob, fineTuningJob.id, undefined, "cancelled");
+
+    await cleanupTestFile(trainingFile.id);
+    await cleanupTestFile(validationFile.id);
+  }
+
+  async function deployJobHelper(
+    completedJobIdEnvVar: string,
+    deploymentFormat: string,
+    deploymentCapacity: number,
+    testPrefix: string,
+  ): Promise<void> {
+    const completedJobId = process.env[completedJobIdEnvVar];
+    if (!completedJobId) {
+      console.warn(`Environment variable ${completedJobIdEnvVar} is not set. Skipping test.`);
+      return;
+    }
+
+    const subscriptionId = process.env["AZURE_AI_PROJECTS_TESTS_AZURE_SUBSCRIPTION_ID"]!;
+    const resourceGroup = process.env["AZURE_AI_PROJECTS_TESTS_AZURE_RESOURCE_GROUP"]!;
+    const projectEndpoint = process.env["AZURE_AI_PROJECTS_TESTS_PROJECT_ENDPOINT"]!;
```

Comment:
We should use `assertEnvironmentVariable` recorder helper to load env vars in tests.